Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/Events/UnityEventDefinitions/TouchInteractionEvent.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// A Unity event with HandTrackingInputEventData. This event is used in the event configuration for the 
+    /// Touch state.
+    /// </summary>
+    [System.Serializable]
+    public class TouchInteractionEvent : UnityEvent<HandTrackingInputEventData> { }
```

Comment:
Some of our other interaction events finally started including information like what object was the original target of the event as well as the pointer that fired the event. I noticed that the HandTrackingInputEventData only includes the controller, not the pointer that fired the event. It also doesn't include the original target of the event. Consider adding those for this event, or to be honest in BaseInputEventData. Might want to just make this a bug to fix later.

Reply:
Added this as a new item: #8955   

It is possible to retrieve the targeted game object by getting the poke pointer's target:

```
       TouchEvents touchEvents = interactiveElement.GetStateEvents<TouchEvents>("Touch");

        touchEvents.OnTouchStarted.AddListener((touchData) =>
        {
            foreach (var pointer in touchData.InputSource.Pointers)
            {
                if (pointer is PokePointer)
                {
                    Debug.Log(pointer.Result.CurrentPointerTarget.name);
                }
            }
        });
```
But we have to loop through the list of pointers.

It is much easier to get the current pointer and the game object with the FocusEventData:

```
        FocusEvents focusEvents = interactiveElement.GetStateEvents<FocusEvents>("Focus");

        focusEvents.OnFocusOn.AddListener((focusData) =>
        {
            Debug.Log(focusData.Pointer.PointerName);
            Debug.Log(focusData.NewFocusedObject);
        });
```